File: TEST data for large repo/repo_Copy-2/Blueprism/BluePrism.Setup/ApiVersion.cs
using System;

namespace BluePrism.Setup
{
    public class ApiVersion
    {
        public Version Version { get; set; }
        public string ApiHash { get; set; }

        public ApiVersion(string apiVersion)
        {
            try
            {
                var split = apiVersion.Split('-');
                Version = Version.Parse(split[0]);
                if (split.Length > 1)
                    ApiHash = split[1];
            }
            catch
            {
                Version = null;
                ApiHash = apiVersion;
            }
        }

        public static ApiVersion Parse(string apiVersion)
        {
            return new ApiVersion(apiVersion);
        }

        public override string ToString()
        {
            if (Version == null) return ApiHash;
            if (string.IsNullOrEmpty(ApiHash)) return Version.ToString();
            return $"{Version}-{ApiHash}";
        }

        public string ToFriendlyString()
        {
            if (Version == null) return ApiHash;
            return $"{Version.Major}.{Version.Minor}.{Version.Build}";
        }

        public static bool AreCompatible(string a, string b)
        {
            return AreCompatible(Parse(a), Parse(b));
        }

        public static bool AreCompatible(ApiVersion a, ApiVersion b)
        {
            return a.Version.Major == b.Version.Major && a.Version.Minor == b.Version.Minor && a.ApiHash == b.ApiHash;
        }
    }
}

File: TEST data for large repo/REPO_Copy-6/Blueprism/BluePrism.Server.Domain.Models/ApiVersion.vb
Public Class ApiVersion

    Public Property Version As Version
    Public Property ApiHash As String

    Public Sub New(apiVersion As String)
        Try
            Dim split = apiVersion.Split("-"c)
            Version = Version.Parse(split(0))
            If split.Length > 1 Then
                ApiHash = split(1)
            End If
        Catch 'If anythin goes wrong just store the full string
            Version = Nothing
            ApiHash = apiVersion
        End Try
    End Sub

    Public Shared Function Parse(apiVersion As String) As ApiVersion
        Return New ApiVersion(apiVersion)
    End Function

    Public Overrides Function ToString() As String
        If Version Is Nothing Then Return ApiHash
        If String.IsNullOrEmpty(ApiHash) Then Return Version.ToString
        Return $"{Version}-{ApiHash}"
    End Function

    Public Function ToFriendlyString() As String
        If Version Is Nothing Then Return ApiHash
        Return $"{Version.Major}.{Version.Minor}.{Version.Build}"
    End Function

    Public Shared Function AreCompatible(a As String, b As String) As Boolean
        Return AreCompatible(Parse(a), Parse(b))
    End Function

    Public Shared Function AreCompatible(a As ApiVersion, b As ApiVersion) As Boolean
        Return a.Version.Major = b.Version.Major AndAlso
               a.Version.Minor = b.Version.Minor AndAlso
               a.ApiHash = b.ApiHash
    End Function

End Class

